### DISCUSSION TITLE
Physical angle at triple point

### DISCUSSION THREAD

**Discussion by gautierromain** [mtex-toolbox/mtex#1377] [Ask Anything]
Hello everyone,
I am using Mtex and studying triple points and more specifically the relation between orientation and the physical angle at the junction.
Thus I want to know if it is possible to know the physical angle and the orientation of a grain at the triple point. To be more specific, here is an exemple :

tp = grains.triplePoints('Phase_1') 
tp(1).angles

The triple point 1 is considering grains 1, 2 and 3 and tp(1).angles is giving me the 3 angles around the triple points. However I don't understand how these angles are sorted, how can I link them to my grains ?

Thanks in advance for your help.

Cheers 

Romain

**Comment by kilir** [comment]
Hi,
you can address the grains and boundaries adjacent to a triple point by id.
Example:
```
mtexdata titanium
grains = ebsd.calcGrains;
grains = smooth(grains,10);

tp = grains.triplePoints;

n=13

tpl =  tp(n).grainId;
for i=1:3
plot(grains('id',tpl(i)))
text(grains('id',tpl(i)),num2str(i))
hold on
end
hold on
plot(tp(n),'MarkerSize',20)

% add angles
plot(grains.boundary('id',tp(n).boundaryId),tp(n).angles/degree,'linewidth',3)
hold off
```
Hope that helps,
Rüdiger

**Reply by gautierromain** [reply]
Thank you for your quick answer ! 
This is great, however I have forget to mention that I want to study it for a large amount of data. Thus I am not sure that reading the angles on EBSD map is the quickest way to do it.

Isn't it a way to link the physical angle to the grainId ? Like tp(1).angles.grainId ?

Thanks again for your answer

Romain

**Reply by kilir** [reply]
Of course this isn't meant to read off the angles, however just a simple demonstration how grainId and angles are linked. A set of triple points has a list of nx3 angles and a list of corresponding nx3 grainIds.

Cheers,
Rüdiger

**Reply by gautierromain** [reply]
I am not sure to understand. If we take your code as exemple, it gives me the image below
![tp1](https://user-images.githubusercontent.com/89841030/158166770-ab42a448-5424-416d-9266-925f2d940fb5.jpg)

When I write tp(1).angles\degree, I get 

ans =

  105.4955  114.6554  139.8491

As we can see here the first angle do not correspond to the first grain. The first grain has a physical angle of 139.8° (which is the third angle). So it seems that there is no link between angles and grainId ...

Maybe another solution exists or maybe I didn't understand what you were explaining me ?

Romain
